public/API/API_GetUserSetRequests.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ *  API_GetUserSetRequests - gets user's set request list
+ *    u : username
+ *
+ *  array      RequestedSets
+ *    object     [value]
+ *      int        GameID                     unique identifier of the game
+ *      string     Title                      set title
+ *      int        ConsoleID                  the id of the console that the set is for
+ *      string     ConsoleName                the name of the console that the set is for
+ *      string     ImageIcon                  the url of the sets icon
+ *  int        TotalRequests              maximum number of requests a user has
+ *  int        PointsForNext              number of points remaining until maximum request increase
+ */
+
+use App\Models\Game;
+use App\Models\User;
+use App\Models\UserGameListEntry;
+
+$user = User::firstWhere('User', request()->query('u'));
+if (!$user) {
+    return response()->json([]);
+}
+
+$requestedSets = UserGameListEntry::where('user_id', $user->ID)
+    ->where('type', 'achievement_set_request')
+    ->pluck('GameID')
+    ->toArray();
+
+$userRequestInfo = UserGameListEntry::getUserSetRequestsInformation($user);
+
+$requestList = [];
+foreach ($requestedSets as $gameId) {
+    $gameData = Game::with('system')->find($gameId);
+    $requestList[] = [
+        'GameID' => $gameData['ID'],
+        'Title' => $gameData['Title'],
+        'ConsoleID' => $gameData['ConsoleID'],
+        'ConsoleName' => $gameData->system['Name'],
```

Comment:
This kicks off 2 SQL queries for each of the user's set requests. Against some users in the database, this results in 500+ queries every time this endpoint gets called.

Reply:
As it stands none of the game data is fetched by the UserGameListEntry Model, it only provides the GameID from the SetRequests table which I then have to look up in order to get the data for the response.

Is the second SQL query that you are referring to the one for ConsoleName? The name is not stored in the GameData table so I have to call out to the Console table for it, I can remove that if it adds too much weight to the request.

Reply:
The two queries are coming from these two lines:
```php
$gameData = Game::with('system')->find($gameId);

'ConsoleName' => $gameData->system['Name'],
```
If you want to see the queries firsthand, in _.env_, make sure this value is set:
```
DEBUGBAR_ENABLED=true
```
Then copy over most of your endpoint code to some random PHP page and use hardcoded values for the user. The debugbar will show all the SQL queries that get executed on the page - you will see two queries kicked off for every game on the user's set request list.

The [issue](https://www.appdynamics.com/blog/product/common-application-problems-and-how-to-fix-them-the-select-n-1-problem/) is the queries are being kicked off in the loop. You'll need to lift this data fetching out of the loop to resolve the performance issue.

Reply:
I have lifted the queries out of the loop, after testing on a PHP page like you suggested each query is now only run once.

Reply:
Nice, I’ll use that method in the future. I was using Content.php but that has a lot of other calls on each page.